### ISSUE TITLE
Importing ROOT prevents Python garbage collection

### ISSUE THREAD

**Issue by fwinkl** [root-project/root#16402]
### Check duplicate issues.

- [X] Checked for duplicates

### Description

In ATLAS we noticed a strange behavior of some of our python scripts after switching to the latest ROOT version.

Importing ROOT seems to prevent proper garbage collection in Python, i.e. the class destructor `__del__` is not being called before the process exits.

Reproducible with ROOT 6.32.00 to 6.32.04, i.e. the problem was introduced at some point after 6.30.08.

Internal Jira issues: [ADHI-4985](https://its.cern.ch/jira/browse/ADHI-4985) [ATR-29707](https://its.cern.ch/jira/browse/ATR-29707) 

### Reproducer

On lxplus9 with ROOT 6.32.04 consider this python script (thanks to Wainer Vandelli):
```python
import ROOT

class Test(object):
     def __init__(self):
             print('Init')
     def __del__(self):
             print('Del')

obj = Test()
```
```sh
> python3 --version
Python 3.9.18
> python3 test.py 
Init
> 
```
Remove the `import ROOT` and the result is as expected:
```sh
> python3 test.py
Init
Del
```
Explicitly calling `del obj` can be used as a workaround but that's certainly not what the user expects having to do.

### ROOT version

ROOT 6.32.04

### Installation method

cvmfs

### Operating system

Linux

**Comment by guitargeek** [comment]
This is a problem with the cppyy API, as this reproducer is sufficient:
```python
import cppyy

cppyy.include('CPyCppyy/API.h')
cppyy.gbl.CPyCppyy.Scope_Check(cppyy.nullptr)

class Test(object):
     def __init__(self):
             print('Init')
     def __del__(self):
             print('Del')

obj = Test()
```
From ROOT 6.32, we used the CPyCppyy API in the ROOT module. That's why there is this issue.

**Comment by guitargeek** [comment]
A fix in on the way.

**Comment by guitargeek** [comment]
Fixed in 6.34.00 and also I'll backport it to the 6.32 branch so the fix is available in 6.32.06 when it's released.
